From bf6f3137d8838e86ffcd8b568cd741b6a252c1ec Mon Sep 17 00:00:00 2001
From: Kedareswara rao Appana <appana.durga.rao@xilinx.com>
Date: Tue, 22 Mar 2016 12:47:54 +0530
Subject: [PATCH 195/793] vdmatest: xilinx: Fix race condition in the test
 client

Current test client driver preparing the VDMA descriptors
upto the h/w configured frames but submitting only the last
desc which is wrong.

This patch fixes this issue by submitting all the prepared
descriptors.

Signed-off-by: Kedareswara rao Appana <appanad@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
---
 drivers/dma/xilinx/vdmatest.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/dma/xilinx/vdmatest.c b/drivers/dma/xilinx/vdmatest.c
index 4a6b941..516bdc5 100644
--- a/drivers/dma/xilinx/vdmatest.c
+++ b/drivers/dma/xilinx/vdmatest.c
@@ -332,6 +332,7 @@ static int xilinx_vdmatest_slave_func(void *data)
 			xt.frame_size = 1;
 			rxd = rx_dev->device_prep_interleaved_dma(rx_chan,
 								  &xt, flags);
+			rx_cookie = rxd->tx_submit(rxd);
 		}
 
 		for (i = 0; i < frm_cnt; i++) {
@@ -352,6 +353,7 @@ static int xilinx_vdmatest_slave_func(void *data)
 			xt.frame_size = 1;
 			txd = tx_dev->device_prep_interleaved_dma(tx_chan,
 								  &xt, flags);
+			tx_cookie = txd->tx_submit(txd);
 		}
 
 		if (!rxd || !txd) {
@@ -372,12 +374,10 @@ static int xilinx_vdmatest_slave_func(void *data)
 		init_completion(&rx_cmp);
 		rxd->callback = xilinx_vdmatest_slave_rx_callback;
 		rxd->callback_param = &rx_cmp;
-		rx_cookie = rxd->tx_submit(rxd);
 
 		init_completion(&tx_cmp);
 		txd->callback = xilinx_vdmatest_slave_tx_callback;
 		txd->callback_param = &tx_cmp;
-		tx_cookie = txd->tx_submit(txd);
 
 		if (dma_submit_error(rx_cookie) ||
 				dma_submit_error(tx_cookie)) {
-- 
1.7.5.4

